pyro/infer/autoguide/guides.py

```diff
@@ -874,27 +875,31 @@ def _setup_prototype(self, *args, **kwargs):
         super()._setup_prototype(*args, **kwargs)
         # Initialize guide params
         self.loc = nn.Parameter(self._init_loc())
+        self.scale = PyroParam(
+            torch.full_like(self.loc, self._init_scale), self.scale_constraint
+        )
         self.scale_tril = PyroParam(
-            eye_like(self.loc, self.latent_dim) * self._init_scale,
-            self.scale_tril_constraint,
+            eye_like(self.loc, self.latent_dim), self.scale_tril_constraint
         )
 
     def get_base_dist(self):
         return dist.Normal(
-            torch.zeros_like(self.loc), torch.zeros_like(self.loc)
+            torch.zeros_like(self.loc), torch.ones_like(self.loc)
         ).to_event(1)
 
     def get_transform(self, *args, **kwargs):
-        return dist.transforms.LowerCholeskyAffine(self.loc, scale_tril=self.scale_tril)
+        scale_tril = self.scale[..., None] * self.scale_tril
+        return dist.transforms.LowerCholeskyAffine(self.loc, scale_tril=scale_tril)
 
     def get_posterior(self, *args, **kwargs):
         """
         Returns a MultivariateNormal posterior distribution.
         """
-        return dist.MultivariateNormal(self.loc, scale_tril=self.scale_tril)
+        scale_tril = self.scale[..., None] * self.scale_tril
+        return dist.MultivariateNormal(self.loc, scale_tril=scale_tril)
 
     def _loc_scale(self, *args, **kwargs):
-        return self.loc, self.scale_tril.diag()
+        return self.loc, self.scale * self.scale_tril.diag()
```

Comment:
does `scale` need an unsqueeze here?

Reply:
nope, `scale` is already the correct shape

Reply:
oh i see this is only for marginals